model/riscv_insts_vext_mem.sail

```diff
@@ -0,0 +1,1539 @@
+/* ************************************************************************ */
+/* This file implements part of the vector extension.                       */
+/* Chapter 7: vector loads and stores                                       */
+
+/* ************************************************************************ */
+
+mapping nfields_int : bits(3) <-> int = {
+  0b000     <-> 1,
+  0b001     <-> 2,
+  0b010     <-> 3,
+  0b011     <-> 4,
+  0b100     <-> 5,
+  0b101     <-> 6,
+  0b110     <-> 7,
+  0b111     <-> 8
+}
+
+mapping nfields_string : bits(3) <-> string = {
+  0b000     <-> "1",
+  0b001     <-> "2",
+  0b010     <-> "3",
+  0b011     <-> "4",
+  0b100     <-> "5",
+  0b101     <-> "6",
+  0b110     <-> "7",
+  0b111     <-> "8"
+}
+
+mapping vlewidth_bitsnumberstr : vlewidth <-> string = {
+  VLE8      <-> "8",
+  VLE16     <-> "16",
+  VLE32     <-> "32",
+  VLE64     <-> "64"
+}
+
+mapping encdec_vlewidth : vlewidth <-> bits(3) = {
+  VLE8      <-> 0b000,
+  VLE16     <-> 0b101,
+  VLE32     <-> 0b110,
+  VLE64     <-> 0b111
+}
+
+mapping vlewidth_bytesnumber : vlewidth <-> int = {
+  VLE8      <-> 1,
+  VLE16     <-> 2,
+  VLE32     <-> 4,
+  VLE64     <-> 8
+}
+
+mapping vlewidth_pow : vlewidth <-> int = {
+  VLE8      <-> 3,
+  VLE16     <-> 4,
+  VLE32     <-> 5,
+  VLE64     <-> 6
+}
+
+mapping bytes_wordwidth : int <-> word_width = {
+  1 <-> BYTE,
+  2 <-> HALF,
+  4 <-> WORD,
+  8 <-> DOUBLE
+}
+
+/* Vector misaligned checking (used in vload) */
+val vcheck_misaligned : (xlenbits, word_width) -> bool effect {undef}
+function vcheck_misaligned(vaddr : xlenbits, width : word_width) -> bool = {
+  if   plat_enable_misaligned_access() then false
+  else match width {
+    BYTE     => false,
+    HALF     => vaddr[0] == bitone,
+    WORD     => vaddr[0] == bitone | vaddr[1] == bitone,
+    DOUBLE   => vaddr[0] == bitone | vaddr[1] == bitone | vaddr[2] == bitone
+  };
+}
+
+/* *****VLOAD**************VLETYPE(Vector Load Unit-Strided Normal, nf=0, mop=0, lumop=0)*************************** */
```

Comment:
What on earth is going on with this header

Reply:
All headers are modified.